ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT/ChainState.hs

```diff
@@ -236,7 +239,8 @@ invariant (SecurityParam k)
 
     allEbbs p =
       Map.null $
-      Map.filterWithKey (\slot mSlot -> not (p slot mSlot)) (unEbbMap ebbs)
+      Map.filterWithKey (\slot (EbbInfo _ mSlot) -> not (p slot mSlot)) $
+      unEbbMap ebbs
 
```

Comment:
Don't forget about #1357. Run the tests with the `checktvarinvariant` flag on the `io-sim-classes` package to see whether we're still affected by the strange leak.

Reply:
I ran this on this same day you commented; it passed. The `INLINE` pragma I added seems sufficient.